stream-api/src/main/kotlin/fund/cyber/markets/api/trades/TradesBroadcastersIndex.kt

```diff
@@ -12,9 +13,53 @@ class TradesBroadcastersIndex : TradesChannelsIndexUpdateListener {
         index.getOrPut(pair, { HashMap() }).put(exchange, broadcaster)
     }
 
-    fun broadcastersFor(pairs: List<TokensPair>): Collection<TokensPairTradesBroadcaster> {
+    fun broadcastersForPairs (pairs: List<TokensPair>): Collection<TokensPairTradesBroadcaster> {
         return index
                 .filter { (pair, _) -> pairs.contains(pair) }
                 .flatMap { (_, pairIndex) -> pairIndex.values }
     }
+
+    fun broadcastersFor(pairs: List<TokensPair>, exchanges: List<String> ): Collection<TokensPairTradesBroadcaster> {
+        return index
+                .filter { (pair, _) -> pairs.contains(pair) }
+                .flatMap { (_, pairIndex) -> pairIndex.entries }
+                .filter { (exchange, _) -> exchanges.contains(exchange)}
+                .map { (_, broadcaster) -> broadcaster }
+    }
+
+    fun broadcastersForExchanges (exchanges: List<String>): Collection<TokensPairTradesBroadcaster> {
+        return index
+                .flatMap { (_, pairIndex) -> pairIndex.entries }
+                .filter { (exchange, _) -> exchanges.contains(exchange)}
+                .map { (_, broadcaster) -> broadcaster }
+    }
+
+    fun broadcastersForAll (): Collection<TokensPairTradesBroadcaster> {
+        return index
+                .flatMap { (_, pairIndex) -> pairIndex.entries }
+                .map { (_, broadcaster) -> broadcaster }
+    }
+
+    fun broadcastersGetAllPairs (): Collection<TokensPair>{
+        return index.keys
+    }
+
+    fun broadcastersGetExchangesTree (): MutableMap<String, List<TokensPair>>{
```

Comment:
getAllExchangesWithPairs or something like this

exchanges tree is a common definition, it's better to have more concrete one